app/javascript/article-form/articleForm.jsx

```diff
@@ -269,6 +270,17 @@ export default class ArticleForm extends Component {
     });
   };
 
+  toggleModal = () => {
+    if (this.state.edited) {
+      this.setState({
+        showModal: !this.state.showModal,
+      });
+    } else {
+      // If the user has not edited the body we send them home
+      window.location.href = '/';
+    }
```

Comment:
(just a thought / not blocking) - I'm not sure it's for this commit, maybe a future optimisation, but if `this.state.edited` seems to get set to true if a user:

- starts a new post
- navigates away without changes
- starts a new post again
- this time on navigate away without changes I see the modal

Bit of an edge case though, and more to do with where we're setting the edited state than these changes